lib/trino-filesystem/src/main/java/io/trino/filesystem/Locations.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem;
+
+public final class Locations
+{
+    private Locations() {}
+
+    public static String appendPath(String location, String path)
+    {
+        if (!location.endsWith("/")) {
```

Comment:
Shouldn't we verify here that the outcome is a valid path?

Reply:
This is a complicated topic. Locations are not URIs. They actually Hadoop paths, which means that URI characters like `%` are literals, and aren't guaranteed to be valid URI syntax, so you can't parse them as URIs. We talked to @rdblue about this need to add something to the Iceberg specification about URI syntax.

We should really try to handle this holistically, but you raise a good point that the conversation from `new Path(...)` here actually loses safety, since we could have a location with a query string or fragment, and thus simple append is not safe. For now, we could add some simple validation to this method that the location doesn't contain a query string `?` or fragment '#` character.

As part of this project, we should audit all locations where paths can be provided (table metadata, schema location, procedures, table properties, etc.) and do some basic validation, since we've lost that in the conversation away from Hadoop `Path`.

Reply:
> They are actually Hadoop paths

What about using a combination of the code taken from `org.apache.hadoop.fs.Path` constructor and `toString()` method then to obtain the string representation of the hadoop path. Copy and adapt to Trino's needs the `org.apache.hadoop.fs.Path` code.

Do note that this utility method will apply to all data lake connectors ( hive, iceberg, delta, hudi).

Reply:
We actually don't want to normalize paths, since that means tables with things like double slashes, which are valid paths in object stores, would not be readable. It's complicated.

Reply:
I added the check for query string and fragment. We should follow up on this. Most concatenation should use this utility, etc.